dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -221,249 +253,282 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
-                        {
-                            return false;
-                        }
+                        m_pushTriggerRegistration.Unregister(true);
+                        m_pushTriggerRegistration = nullptr;
+                    }
+                }
+            ) };
 
-                        if (name == backgroundTaskFullName)
-                        {
-                            s_pushTriggerRegistration = task.Value();
-                            return true;
-                        }
+            /* Every packaged app is already getting the benefit of COM Registration to handle notifications.
+               However, we would benefit from power management improvements in Background Infrastructure.
+               So we are continuing to register a Push trigger post 19041 builds */
+               
+            if (PushNotificationHelpers::IsPackagedAppScenario())
+            {
+                {
+                    auto lock = m_lock.lock_shared();
+                    THROW_HR_IF(E_INVALIDARG, m_pushTriggerRegistration);
+                }
 
-                        throw winrt::hresult_invalid_argument(L"RegisterActivator has different clsid registered.");
-                    }) };
+                winrt::hstring registeredClsidStr{ winrt::to_hstring(registeredClsid) };
+                winrt::hstring backgroundTaskFullName{ backgroundTaskName + registeredClsidStr };
 
+                bool isTaskRegistered = IsBackgroundTaskRegistered(backgroundTaskFullName);
+               
                 if (!isTaskRegistered)
                 {
+                    BackgroundTaskBuilder builder{ nullptr };
+
                     builder = BackgroundTaskBuilder();
                     builder.Name(backgroundTaskFullName);
 
                     PushNotificationTrigger trigger{};
                     builder.SetTrigger(trigger);
 
-                    THROW_HR_IF(E_NOTIMPL, !AppModel::Identity::IsPackagedProcess());
-
                     // In case the interface is not supported, let it throw.
                     auto builder5{ builder.as<winrt::IBackgroundTaskBuilder5>() };
-                    builder5.SetTaskEntryPointClsid(taskClsid);
-                    winrt::com_array<winrt::IBackgroundCondition> conditions = details.GetConditions();
-                    for (auto condition : conditions)
-                    {
-                        builder.AddCondition(condition);
-                    }
+                    builder5.SetTaskEntryPointClsid(registeredClsid);
+
+                    auto lock{ m_lock.lock_exclusive() };
+                    m_pushTriggerRegistration = builder.Register();
                 }
             }
 
-            BackgroundTaskRegistration registeredTaskFromBuilder{ nullptr };
+            if (AppModel::Identity::IsPackagedProcess())
+            {
+                bool registeredEvent{ false };
+                THROW_HR_IF(E_INVALIDARG, registeredClsid == winrt::guid());
```

Comment:
registeredClsid is not an arg anymore. So this should be:
THROW_HR_IF(E_UNEXPECTED, registeredClsid == winrt::guid());